### PR TITLE
Generalizes the chemical blacklist, refactors some other chemical lists and cleans up some related code

### LINKED ISSUE #11857: Multiply any chem by using humanoid bodies as beakers
**Problem Description**:
You can basically multiply chems inside humanoids by filling them up with the chem and then filling them up with Phosphorus, Sugar and Potassium to smoke the chemical. The humanoid will then release the chems they had in their body as smoke but the chems in their body don't get used up so the chems double inside their body because they still have the chems inside them and inhale the chems they should've smoked out.

**What did you expect to happen**: 
I wanted to fart out a saline glucose cloud from my body.

**What happened instead**:
I farted out a saline glucose smoke but the chems inside my body doubled for some reason and everyone who was in range also got the same amount of saline into their body.

**Why is this bad/What are the consequences:**
Unlimited rare chems for example intrioprodril.

**Steps to reproduce the problem**:
1. Inject yourself (or another humanoid) with any chemical.
2. Drink (or force feed it to the humanoid) 10u potassium, 10u phosphorus and 10u sugar.
3. ???
4. Profit. You now doubled the initial amount of chemicals inside your body.

**When did the problem start happening**:
It still works.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?)**:
:)

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do:
Resolves #11857 

- Generalizes `can_synth` into a single global list, making it easier to see what chemicals you can and cannot replicate.
- All instances of `can_synth` have been replaced by a check for the global list.
- Slightly cleans up bee code and give it an error when injecting the queen instead of when you're trying to add the queen to the apiary
- Smoke now can't carry smoke powder with it.
- Removes the flying potion reagent because it's 99% commented out and would require an entire rewrite anyways if someone wanted to revive it.
- Makes it so sleepers (and IV bags) can still dialysis out the following chems, but returning "Saturated activated charcoal" instead, but still removing the chemical from the person's bloodstream:
 ("polonium", "initropidril", "concentrated_initro", "sodium_thiopental", "ketamine", "adminordrazine", "nanites", "hell_water", "mutationtoxin", "amutationtoxin", "venom", "spore", "stimulants", "stimulative_agent", "syndicate_nanites", "ripping_tendrils", "boiling_oil", "envenomed_filaments", "lexorin_jelly", "kinetic", "cryogenic_liquid", "liquid_dark_matter", "b_sorium", "reagent", "dragonsbreath", "nanocalcium", "xenomicrobes", "nanomachines", "gibbis", "prions", "spidereggs", "heartworms", "bacon_grease", "fungalspores", "jagged_crystals", "salmonella", "lavaland_extract", "stable_mutagen", "beer2", "curare", "gluttonytoxin",  "smoke_powder")

## Why It's Good For The Game
Refactors some unused global lists into being used, cleans up code around it.
Fixes a way to (easily) dupe chemicals.

## Testing
Tried to dialysis out some blocked chems, ~~didn't work and my spessman died anyways~~ got "????" instead and my spessman survived :D
Queen bees don't accept blocked chems
An Odysseus doesn't accept blocked chems in the syringe gun

## Changelog
:cl:
tweak: Some chemicals will give "Saturated activated charcoal" instead of the chemical itself when being dialysis'ed
/:cl:

### CODE REVIEW

**Comment by SpringSkipper** [comment]
Being unable to remove some of these chems by dialysis seems like a problem, as that means there is **zero** way to remove them from a dead person - you're stuck defibbing them repeatedly waiting for the charcoal/pentetic to work.

You could have them transform when removed from the body? To like, 'sludge' or something. Dialysis is too important a medical tool to just shrug it away like this in my opinion.

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 1]
```diff
@@ -290,30 +290,32 @@
...
+	var/obj/item/reagent_containers/syringe/S = I
+	if(S.reagents.has_reagent("royal_bee_jelly")) //checked twice, because I really don't want royal bee jelly to be duped
+		if(S.reagents.has_reagent("royal_bee_jelly", 5))
+			S.reagents.remove_reagent("royal_bee_jelly", 5)
+			var/obj/item/queen_bee/qb = new(user.drop_location())
+			qb.queen = new(qb)
+			if(queen && queen.beegent)
+				qb.queen.assign_reagent(queen.beegent) //Bees use the global singleton instances of reagents, so we don't need to worry about one bee being deleted and her copies losing their reagents.
+			user.put_in_active_hand(qb)
+			user.visible_message("<span class='notice'>[user] injects [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>","<span class ='warning'>You inject [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>")
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:304]:
```suggestion
			user.visible_message("<span class='notice'>[user] injects [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>","<span class='warning'>You inject [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>")
```

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 2]
```diff
@@ -290,30 +290,32 @@
...
-				if(queen && queen.beegent)
-					qb.queen.assign_reagent(queen.beegent) //Bees use the global singleton instances of reagents, so we don't need to worry about one bee being deleted and her copies losing their reagents.
-				user.put_in_active_hand(qb)
-				user.visible_message("<span class='notice'>[user] injects [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>","<span class ='warning'>You inject [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>")
-			else
-				to_chat(user, "<span class='warning'>You don't have enough royal bee jelly to split a bee in two!</span>")
+	if(!istype(I, /obj/item/reagent_containers/syringe))
+		return ..()
+	var/obj/item/reagent_containers/syringe/S = I
+	if(S.reagents.has_reagent("royal_bee_jelly")) //checked twice, because I really don't want royal bee jelly to be duped
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:296]: :D

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 3]
```diff
@@ -25,6 +25,10 @@
 	var/printing = FALSE
 	var/static/list/pill_bottle_wrappers
 	var/static/list/bottle_styles
+	var/list/safe_chem_list = list("antihol", "charcoal", "epinephrine", "insulin", "teporone","silver_sulfadiazine", "salbutamol",
+									"omnizine", "stimulants", "synaptizine", "potass_iodide", "oculine", "mannitol", "styptic_powder",
+									"spaceacillin", "salglu_solution", "sal_acid", "cryoxadone", "blood", "synthflesh", "hydrocodone",
+									"mitocholide", "rezadone", "menthol")
```
Comment (Henri215) [inline code/modules/reagents/chemistry/machinery/chem_master.dm:31]:
```suggestion
	var/list/safe_chem_list = list("antihol", "charcoal", "epinephrine", "insulin", "teporone", "silver_sulfadiazine", "salbutamol",
									"omnizine", "stimulants", "synaptizine", "potass_iodide", "oculine", "mannitol", "styptic_powder",
									"spaceacillin", "salglu_solution", "sal_acid", "cryoxadone", "blood", "synthflesh", "hydrocodone",
									"mitocholide", "rezadone", "menthol")
```

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 4]
```diff
@@ -17,6 +17,10 @@
 	slot_flags = SLOT_BELT
 	var/ignore_flags = FALSE
 	var/safety_hypo = FALSE
+	var/static/list/safe_chem_list = list("antihol", "charcoal", "epinephrine", "insulin", "teporone","silver_sulfadiazine", "salbutamol",
+									"omnizine", "stimulants", "synaptizine", "potass_iodide", "oculine", "mannitol", "styptic_powder",
+									"spaceacillin", "salglu_solution", "sal_acid", "cryoxadone", "blood", "synthflesh", "hydrocodone",
```
Comment (Henri215) [inline code/modules/reagents/reagent_containers/hypospray.dm:22]:
```suggestion
	var/static/list/safe_chem_list = list("antihol", "charcoal", "epinephrine", "insulin", "teporone", "silver_sulfadiazine", 
"salbutamol",
									"omnizine", "stimulants", "synaptizine", "potass_iodide", "oculine", "mannitol", "styptic_powder",
									"spaceacillin", "salglu_solution", "sal_acid", "cryoxadone", "blood", "synthflesh", "hydrocodone",
```

File: code/modules/mob/living/simple_animal/hostile/bees.dm (hunk 2 above)
Comment (DGamerL) [reply code/modules/mob/living/simple_animal/hostile/bees.dm:296]: I love oldcoders, I'll remove the funny check

File: code/_globalvars/lists/reagents_lists.dm [hunk 5]
```diff
@@ -52,20 +52,18 @@ GLOBAL_LIST_INIT(drinks, list("beer2","hot_coco","orangejuice","tomatojuice","li
 //Liver Toxins list
 GLOBAL_LIST_INIT(liver_toxins, list("toxin", "plasma", "sacid", "facid", "cyanide","amanitin", "carpotoxin"))
 
-//Random chem blacklist
+//General chem blacklist // This is for the really good stuff that we just want to restrict for things like bees and smoke
```
Comment (hal9000PR) [inline code/_globalvars/lists/reagents_lists.dm:55]:
```suggestion
//General chem blacklist // This is for the really good stuff that we just want to restrict from things like bees and smoke
```

**Review by hal9000PR — CHANGES_REQUESTED** [review]

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**